docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,740 @@
+goods as they move through a supply chain. Records for goods include a
+history of ownership and custodianship, as well as histories for a
+variety of properties such as temperature and location. These
+properties are managed through a user-specifiable system of record
+types.
+
+
+State
+=====
+
+All T&T objects are serialized using Protocol Buffers before being
+stored in state. These objects include: Agents, Properties
+(accompanied by their auxiliary PropertyPage objects), Proposals,
+Records, and RecordTypes. As described in the Addressing_ section
+below, these objects are stored in separate sub-namespaces under the
+T&T namespace. To handle hash collisions, all objects are stored in
+lists within protobuf "Container" objects.
+
+
+Records
+-------
+
+Records represent the goods being tracked by Track and Trade. Almost
+every transaction references some Record.
+
+A Record contains a unique identifier, the name of a RecordType, and
+lists containing the history of its owners and custodians. It also
+contains a ``final`` flag indicating whether further updates can be
+made to the Record and its Properties. If this flag is set to true,
+then no further updates can be made to the Record, including changing
+its ``final`` flag.
+
+.. code-block:: protobuf
+
+   message Record {
+       message AssociatedAgent {
+           // The Agent's public key
+           string agent_id = 1;
+
+           // Approximately when this agent was associated, as a Unix UTC timestamp
+           uint64 timestamp = 2;
+       }
+
+       // The unique user-defined natural key which identifies the
+       // object in the real world (for example, a serial number)
+       string identifier = 1;
+
+       string record_type = 2;
+
+       // Ordered oldest to newest by timestamp
+       repeated AssociatedAgent owners = 3;
+       repeated AssociatedAgent custodians = 4;
+
+       // Flag indicating whether the Record can be updated. If it is set
+       // to true, then the record has been finalized and no further
+       // changes can be made to it or its Properties.
+       bool final = 5;
+   }
+
+
+Note that while information about a Record's owners and custodians are
+included in the object, information about its Properties are stored
+separately (see the Properties_ section below).
+
+Records whose addresses collide will be stored in a list
+alphabetically by identifier.
+
+.. code-block:: protobuf
+
+   message RecordContainer {
+       repeated Record entries = 1;
+   }
+
+.. _Properties:
+
+Properties
+----------
+
+Historical data pertaining to a particular data field of a tracked
+object will be stored as Properties, represented as a list of values
+accompanied by a timestamp and a reporter identifier.
+
+The whole history of updates to Record data is stored in current state
+because this allows for more flexibility in writing transaction rules.
+For example, there might be a rule that no fish can be exchanged whose
```

Comment:
Because the FishNet app isn't described in the spec (vs the design doc), this first fish example could use a bit more info, such as your suggestion: 

     For example, in a fish track-and-trade system, there might be a rule that no fish can be exchanged ...